client/client.go

```diff
@@ -482,3 +482,19 @@ func (w *WrapperV2) SetCredentials(organizationID string, addCredentialsRequest
 
 	return response, nil
 }
+
+// GetCluster fetches details on a cluster using the V2 client.
+func (w *WrapperV2) GetClusterStatus(clusterID string, p *AuxiliaryParams) (*clusters.GetClusterStatusOK, error) {
+	params := clusters.NewGetClusterStatusParams().WithClusterID(clusterID)
+	err := setParamsWithAuthorization(p, w, params)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	response, err := w.gsclient.Clusters.GetClusterStatus(params, nil)
+	if err != nil {
+		return nil, clienterror.New(err)
+	}
+
+	return response, nil
+}
```

Comment:
Wouldn't it be nice if this function could do the unmarshal/marshal dance and return the usable struct instead of the not so usable map[string]interface?

Reply:
In other words, should we move https://github.com/giantswarm/gsctl/blob/scaling-update/commands/scale_cluster.go#L175-L188 over here?

Reply:
I tried this, but then we get a weird situation:
Everything in `client.go` returns the swagger generated types, but now one function would return the apiextensions type. It would then also no longer be `response.Payload` but just `response`. It seems equally inconsistent than just doing it in this function.

I'll move it if it's bothering you, but I don't see a big benefit.